Clover/app/src/main/java/org/floens/chan/core/repository/ImportExportRepository.java

```diff
@@ -295,7 +294,20 @@ private void writeSettingsToDatabase(@NonNull ExportedAppSettings appSettingsPar
         ChanSettings.deserializeFromString(appSettingsParam.getSettings());
     }
 
-    private ExportedAppSettings onUpgrade(int version, ExportedAppSettings appSettings) {
+    @NonNull
+    private ExportedAppSettings onUpgrade(
+            int version, ExportedAppSettings appSettings) throws UpgradeImpossibleException {
+        if (version < 2) {
+            ///////////////////////////////////////
+            // Upgrade from version 1 to version 2.
+            // In version 2 we had to change the way the settings are serialized to a file because
+            // it was not considering the UTF-8 characters. This breaks everything. So we have no other
+            // way other than clear everything.
```

Comment:
"other than not import anything"

Reply:
This is entirely unnecessary because the exception being thrown will be in line 151. The actual JSON fails to parse because of mismatched quotes, so you'd never even get to this point. There's no need to increment the version number.